src/Domain/CoordinateMaps/UniformCylindricalSide.cpp

```diff
@@ -0,0 +1,752 @@
+  // jac(2,1)=R1 sin(theta1)
+  get<2, 1>(jac) = sqrt(
+      square(radius_one_) -
+      square(z_plane_minus_one_ - center_one_[2] +
+             0.5 * (zbar + 1.0) * (z_plane_plus_one_ - z_plane_minus_one_)));
+  // jac(0,2)=cot theta1/rho
+  get<0, 2>(jac) =
+      (z_plane_minus_one_ - center_one_[2] +
+       0.5 * (zbar + 1.0) * (z_plane_plus_one_ - z_plane_minus_one_)) /
+      (get<2, 1>(jac) * get<2, 2>(jac));
+  // jac(2,1)=R1 sin(theta1)/rho^3
+  get<2, 1>(jac) /= cube(get<2, 2>(jac));
+  // jac(2,0)=R2 sin(theta2)
+  get<2, 0>(jac) = sqrt(
+      square(radius_two_) -
+      square(z_plane_minus_two_ - center_two_[2] +
+             0.5 * (zbar + 1.0) * (z_plane_plus_two_ - z_plane_minus_two_)));
+  // jac(1,2)=cot theta2/rho
+  get<1, 2>(jac) =
+      (z_plane_minus_two_ - center_two_[2] +
+       0.5 * (zbar + 1.0) * (z_plane_plus_two_ - z_plane_minus_two_)) /
+      (get<2, 0>(jac) * get<2, 2>(jac));
+  // jac(2,0)=R2 sin(theta2)/rho^3
+  get<2, 0>(jac) /= cube(get<2, 2>(jac));
+  // jac(1,1)=lambda rho^2
+  get<1, 1>(jac) = square(get<2, 2>(jac)) * (get<2, 2>(jac) - 1.0);
+
+  // Now fill Jacobian values
+  get<0, 0>(jac) =
+      square(ybar) * get<2, 1>(jac) +
+      (get<1, 1>(jac) + square(xbar)) * (get<2, 0>(jac) - get<2, 1>(jac)) +
+      xbar * (center_two_[0] - center_one_[0]) / get<2, 2>(jac);
+  get<1, 1>(jac) =
+      square(xbar) * get<2, 1>(jac) +
+      (get<1, 1>(jac) + square(ybar)) * (get<2, 0>(jac) - get<2, 1>(jac)) +
+      ybar * (center_two_[1] - center_one_[1]) / get<2, 2>(jac);
+  get<0, 1>(jac) = xbar * ybar * (get<2, 0>(jac) - 2.0 * get<2, 1>(jac)) +
+                   ybar * (center_two_[0] - center_one_[0]) / get<2, 2>(jac);
+  get<1, 0>(jac) = xbar * ybar * (get<2, 0>(jac) - 2.0 * get<2, 1>(jac)) +
+                   xbar * (center_two_[1] - center_one_[1]) / get<2, 2>(jac);
+  get<1, 2>(jac) =
+      0.5 * (get<0, 2>(jac) * (z_plane_plus_one_ - z_plane_minus_one_) *
+                 (get<2, 2>(jac) - 2.0) +
+             get<1, 2>(jac) * (z_plane_plus_two_ - z_plane_minus_two_) *
+                 (1.0 - get<2, 2>(jac)));
+  get<0, 2>(jac) = get<1, 2>(jac) * xbar;
+  get<1, 2>(jac) *= ybar;
+  get<2, 1>(jac) = (z_plane_minus_two_ - z_plane_minus_one_ +
+                    0.5 * (zbar + 1) *
+                        (z_plane_plus_two_ - z_plane_minus_two_ -
+                         z_plane_plus_one_ + z_plane_minus_one_)) /
+                   get<2, 2>(jac);
+  get<2, 0>(jac) = xbar * get<2, 1>(jac);
+  get<2, 1>(jac) *= ybar;
+  get<2, 2>(jac) =
+      0.5 * ((2.0 - get<2, 2>(jac)) * (z_plane_plus_one_ - z_plane_minus_one_) +
+             (get<2, 2>(jac) - 1.0) * (z_plane_plus_two_ - z_plane_minus_two_));
+  return jac;
+}
+
+template <typename T>
+tnsr::Ij<tt::remove_cvref_wrap_t<T>, 3, Frame::NoFrame>
+UniformCylindricalSide::inv_jacobian(
+    const std::array<T, 3>& source_coords) const {
+  return determinant_and_inverse(jacobian(source_coords)).second;
+}
```

Comment:
I'm just wondering: after all the checking on different ways the inverse map might not be invertible, how come we never have to worry about the Jacobin matrix being invertible? Can you explain it like this: determinant can't be zero unless one mapped coord not affected by any unmapped coords, something clearly pathological that won't happen in real life?

Reply:
Jacobian is taken care of.  Think about the 1d case which is easier.  If you have a map `X(x)`, then invertibility means the inverse map `x(X)` is a function, i.e. it is single-valued.  But the single-valued condition is equivalent to `x(X)` is monotonically increasing or decreasing, which is the same thing as the inverse Jacobian `dx/dX` being always the same sign, i.e. `dx/dX` is never zero.  And that is the same condition that the Jacobian `dX/dx` is finite.